src/util/simplify_expr2.cpp

```diff
@@ -731,6 +740,23 @@ expr2tc neg2t::do_simplify() const
     }
     return constant_vector2tc(value->type, std::move(members));
   }
+  if (is_unsignedbv_type(value))
+  {
+    // Get bit-width of the unsigned type
+    const unsigned int width = value->type->get_width();
+
+    // Compute modulus: 2^width
+    const BigInt modulus = BigInt(1) << width;
+    const expr2tc modulus_expr = constant_int2tc(value->type, modulus);
+
+    // Perform modular negation: (modulus - x) % modulus
+    const expr2tc negated_value = sub2tc(value->type, modulus_expr, value);
+    expr2tc wrap = modulus2tc(value->type, negated_value, modulus_expr);
+    wrap->simplify();
+
+    return wrap;
+  }
+
```

Comment:
This should now be unnecessary.

Reply:
I'm not a big fan of having similar code in two places.
Having the code in `simplify_arith_1op` is more future proof than having the code in the neg2t class.
But I don't feel strongly about this.

It just made more sense for me to copy the existing structure from binary ops also for unary ops and to not specialcase truncation/overflow for specific operators, because this is also not the case elsewhere.